### ISSUE TITLE
Crossover Plots: #Interactions on X vs Cumulative Time on Y

### ISSUE THREAD

**Issue by matthewhammer** [plum-umd/adapton.ocaml#28]
* Kyle, Mike and Matthew discussed this a couple of weeks ago
 * Plot axis: #Interactions on X vs Cumulative Time on Y
 * Plot lines: From-scratch versus Incremental (Nominal and Structural)
 * These plots are computable from our current dataset, for each benchmark that we currently have.

**Comment by kyleheadley** [comment]
If I recall, we were looking for the crossover of (X-initial + X-incremental * interactions = Y-initial + Y-incremental * interactions)

which gives us: crossover-interactions = (Y-initial - X-inital)/(X-incremental - Y-incremental)

I'll add this to the next set of data I gather, and we can see how it looks.

**Comment by matthewhammer** [comment]
@kyleheadley: My use of X and Y referred to the axis of the plot.

What are the "X" and "Y" in your comment above referring to?  The two versions (incremental and non-incremental)?

**Comment by kyleheadley** [comment]
Yes, sorry for the confusion. Did you want to plot unique interactions? The way I remember it, we were 'extending' a single interaction, to generate a crossover point for each test.

I was just going to include it as a data item in our table instead of overhead.

----- Reply message -----

**Comment by matthewhammer** [comment]
Yes, after each initial run, we want to "extend" a single interaction sequence, by imagining that we repeat some interaction type over and over.  We get two lines from that, one for the incremental version, and one for `lazyrecalc`, or whatever is modeling the always-do-it-from-scratch version.

Presenting this in place of overhead in a table sounds good to me.  If things go well, the cross-over and overhead are nearly the same number, as we've discussed before with Mike.

Plots:
----------
In addition to the table column for "Crossover", it'd also be great to see plots for each benchmark, as we discussed with Mike.  I like seeing these, since they meaningfully compare a lot of data: both versions' initial times, as well as both versions' edit+update times.  I think they'll be useful to us for diagnostics, as well as communicating the results in talks.  For example, I'm giving a talk at CMU in a few weeks, and I'd like to show some cross-over plots to summarize all of this numerical data in just a few pictures.

For those plots, the X and Y axis are as described above.

Ideally we'd see **lots** of plots, for each benchmark, for each edit type, there'd be a cross over plot.

Maybe Nick knows how to do this in R easily, to avoid lots of busywork in Excel?  I'm not sure.
